homeassistant/components/actronair_neo/climate.py

```diff
@@ -0,0 +1,263 @@
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the target temperature."""
+        return self._status.user_aircon_settings.temperature_setpoint_cool_c
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature that can be set."""
+        return self._status.min_temp
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature that can be set."""
+        return self._status.max_temp
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set a new fan mode."""
+        api_fan_mode = FAN_MODE_MAPPING.get(fan_mode.lower())
+        await self._status.user_aircon_settings.set_fan_mode(api_fan_mode)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set the HVAC mode."""
+        ac_mode = HVAC_MODE_MAPPING_REVERSE.get(hvac_mode)
+        await self._status.ac_system.set_system_mode(ac_mode)
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set the temperature."""
+        temp = kwargs.get("temperature")
+        await self._status.user_aircon_settings.set_temperature(temperature=temp)
+
+    async def async_turn_on_continuous(self, continuous: bool) -> None:
+        """Set the continuous mode."""
+        await self._status.user_aircon_settings.set_continuous_mode(enabled=continuous)
+
+
+class ActronZoneClimate(
+    CoordinatorEntity[ActronNeoDataUpdateCoordinator], ClimateEntity
+):
+    """Representation of a zone within the Actron Air system."""
+
+    _attr_has_entity_name = True
+    _attr_fan_modes = ["auto", "low", "medium", "high"]
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.TURN_ON
+        | ClimateEntityFeature.TURN_OFF
+    )
+    _attr_entity_registry_enabled_default = True
+
+    def __init__(
+        self,
+        coordinator: ActronNeoDataUpdateCoordinator,
+        serial_number: str,
+        zone: ActronAirNeoZone,
+    ) -> None:
+        """Initialize an Actron Air Neo unit."""
+        super().__init__(coordinator)
+        self._serial_number: str = serial_number
+        self._zone_id: int = zone.zone_id
+        self._attr_name: None = None
+        self._attr_unique_id: str = f"{serial_number}_zone_{zone.zone_id}"
+        self._attr_device_info: DeviceInfo = DeviceInfo(
+            identifiers={(DOMAIN, self._attr_unique_id)},
+            name=zone.title,
+            manufacturer="Actron Air",
+            model="Zone",
+            suggested_area=zone.title,
+        )
+
+    @property
+    def _zone(self) -> ActronAirNeoZone:
+        """Get the current zone data from the coordinator."""
+        status = self.coordinator.get_status(self._serial_number)
+        for zone in status.remote_zone_info:
+            if zone.zone_id == self._zone_id:
+                return zone
+        return None
```

Comment:
I would strongly recommend to update the library to return the zones as `dict[str, ActronAirNeoZone]`. This way you don't have to do a loop every time but just once when creating the dict